### ISSUE TITLE
SQLLab - Saving a query with the same label results in a new query being created

### ISSUE THREAD

**Issue by paulvic** [apache/incubator-superset#5804]
Make sure these boxes are checked before submitting your issue - thank you!

- [x] I have checked the superset logs for python stacktraces and included it here as text if any
- [x] I have reproduced the issue with at least the latest released version of superset
- [x] I have checked the issue tracker for the same issue and I haven't found one similar

### Superset version
0.25.6

### Expected results
When you open a saved query in SQLLab and then save it with the same label, the existing query is updated.

### Actual results
When you open a saved query in SQLLab and then save it with the same label, a new query is created.

### Steps to reproduce
1. Open SQLLab, write a query
2. Save the query with the label "test1"
3. Modify the query and save with the same label "test1"
4. Open saved queries

Results:
There are two queries `test1` saved queries listed.

**Comment by fabiomendescom** [comment]
I am taking this one. More complex than it is seems. No provision for REST update call in the component, React components not receiving the id (and additionally the description). This causes the create to always occur which explains the dup. On it.

**Comment by fabiomendescom** [comment]
I have fixed this bug and added all appropriate tests. However, it is necessary to change superset ui connection. See this link for more info -> https://github.com/apache-superset/superset-ui/issues/104. I have submitted this ticket and a PR with the fix. Will need to wait for this fix to be applied there in order to send a PR for this one. As soon as applied there, I will send a PR for this fix.
